pkg/agent/multicast/mcast_discovery.go

```diff
@@ -134,13 +141,28 @@ func (s *IGMPSnooper) validate(event *mcastGroupEvent) (bool, error) {
 	return true, nil
 }
 
-func (s *IGMPSnooper) validatePacketAndNotify(event *mcastGroupEvent) {
-	allow, _ := s.validate(event)
-	if allow {
-		s.eventCh <- event
+func (s *IGMPSnooper) addToIGMPReportNPStatsMap(item types.McastNPValidationItem, packetLen uint64) {
+	uuid := item.Uuid
+	name := item.Name
+	ruleType := *item.NPType
+	if s.igmpReportNPStats[ruleType][uuid] != nil {
+		if s.igmpReportNPStats[ruleType][uuid][name] != nil {
+			stats := s.igmpReportNPStats[ruleType][uuid][name]
+			stats.Bytes += packetLen
+			stats.Packets += 1
+		} else {
+			s.igmpReportNPStats[ruleType][uuid][name] = &types.RuleMetric{Bytes: packetLen, Packets: 1}
+		}
+	} else {
+		s.igmpReportNPStats[ruleType][uuid] = make(map[string]*types.RuleMetric)
+		s.igmpReportNPStats[ruleType][uuid][name] = &types.RuleMetric{Bytes: packetLen, Packets: 1}
 	}
 }
 
+func (s *IGMPSnooper) GetStatsMap() map[v1beta2.NetworkPolicyType]map[apitypes.UID]map[string]*types.RuleMetric {
+	return s.igmpReportNPStats
```

Comment:
> Need not locking for the caller to access the map?

Reply:
Updated. Locking is needed.

Reply:
Please read https://github.com/antrea-io/antrea/pull/3449#discussion_r883522206